sw/blas/gemm/src/gemm.h

```diff
@@ -11,11 +11,18 @@
 
 #include "snrt.h"
 
+#pragma once
+
 // Guard to avoid conflict with DNN header file
 // TODO: move this definition to Snitch math library to solve problem
 #ifndef PRECISION_T
 #define PRECISION_T
 typedef enum { FP64 = 8, FP32 = 4, FP16 = 2, FP8 = 1 } precision_t;
+#endif
+
+#ifndef IMPLEMENTATION_T
+#define IMPLEMENTATION_T
+typedef enum { BASELINE, NAIVE, NAIVE_UNROLLED, OPT, OPT_EX } implementation_t;
```

Comment:
I would use a consistent naming, either `ex_opt` or `opt_ex` everywhere (configuration file names, implementation function signature, datagen script). It used to be `ex_opt` based on the function signature.